xbmc/games/controllers/windows/ControllerSelect.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ *  Copyright (C) 2021 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "ControllerSelect.h"
+
+#include "FileItem.h"
+#include "ServiceBroker.h"
+#include "dialogs/GUIDialogSelect.h"
+#include "games/controllers/Controller.h"
+#include "games/controllers/ControllerLayout.h"
+#include "games/controllers/ControllerTypes.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "guilib/LocalizeStrings.h"
+#include "guilib/WindowIDs.h"
+#include "utils/Variant.h"
+#include "utils/log.h"
+
+using namespace KODI;
+using namespace GAME;
+
+CControllerSelect::CControllerSelect() : CThread("ControllerSelect")
+{
+}
+
+CControllerSelect::~CControllerSelect() = default;
+
+void CControllerSelect::Initialize(ControllerVector controllers,
+                                   ControllerPtr defaultController,
+                                   bool showDisconnect,
+                                   std::function<void(ControllerPtr)> callback)
+{
+  // Validate parameters
+  if (!callback)
+    return;
+
+  // Initialize state
+  m_controllers = std::move(controllers);
+  m_defaultController = std::move(defaultController);
+  m_showDisconnect = showDisconnect;
+  m_callback = callback;
+
+  // Create thread
+  Create(false);
+}
+
+void CControllerSelect::Deinitialize()
+{
+  // Stop thread
+  StopThread(true);
+
+  // Reset state
+  m_controllers.clear();
+  m_defaultController.reset();
+  m_showDisconnect = true;
+  m_callback = nullptr;
+}
+
+void CControllerSelect::Process()
+{
+  // Select first controller by default
+  unsigned int initialSelected = 0;
+
+  CGUIComponent* gui = CServiceBroker::GetGUI();
+  if (gui == nullptr)
+    return;
+
+  CGUIWindowManager& windowManager = gui->GetWindowManager();
+
+  auto pSelectDialog = windowManager.GetWindow<CGUIDialogSelect>(WINDOW_DIALOG_SELECT);
+  if (pSelectDialog == nullptr)
+    return;
+
+  CLog::Log(LOGDEBUG, "Controller select: Showing dialog for %u controllers", m_controllers.size());
```

Comment:
Please update to the current log standards using `{}`.

For more information please see the [libfmt documentation](https://fmt.dev/latest/syntax.html)